src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,206 @@
+    powerSet = unique(sort.(Oscar.subsets(tempGrdSet,size)))
+
+    setsOfSize = filter(x->length(x)==size,sets)
+    nonSets = setdiff(powerSet,setsOfSize) 
+
+    for set in setsOfSize
+      append!(b[size],collect(Oscar.permutations(set)))
+    end
+    for nonset in nonSets
+      append!(nb[size],collect(Oscar.permutations(nonset)))
+    end
+    for set in b[size]
+      for nonset in nb[size]
+        rel = []
+        for i in 1:size
+          push!(rel,(set[i],nonset[i]))
+        end
+        push!(rels[size],rel)
+        rel = []
+        for i in 1:size
+          push!(rel,(nonset[i],set[i]))
+        end
+        push!(rels[size],rel)
+      end
+    end
+  end
+
+  rels = unique.(rels)
+  return Vector{Vector{Tuple{Int,Int}}}(reduce(vcat,rels))
+end
+
+@doc raw"""
+    quantum_automorphism_group(M::Matroid, structure::Symbol=:bases)
+
+Get the Ideal that defines the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> M = uniform_matroid(3,4);
+
+julia> qAut = quantum_automorphism_group(M,:bases);
+
+julia> length(gens(qAut));
+2040
+```
+"""
+function quantum_automorphism_group(
+  M::Matroid,
+  structure::Symbol=:bases)
+
+  @req matroid_groundset(M) isa Vector{Int} "Matroid must have integer groundset"
+
+  relation_indices = _quantum_automorphism_group_indices(M,structure)
+  n = length(M)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n,n)),[2,1])
+
+  new_relations = []
+
+  for relation in relation_indices
+    temp = one(A)
+    for gen in relation
+      temp = temp * u[gen[1], gen[2]]
+    end
+    push!(new_relations,temp)
+  end
+  return ideal([gens(SN)..., new_relations...])
+
+end
+
+@doc raw"""
+    quantum_automorphism_group(G::Graph)
+
+Get the Ideal that defines the quantum automorphism group of a graph.
+
+# Examples
+```jldoctest
+julia> G = graph_from_edges([[1,2],[2,4]]);
+
+julia> qAut = quantum_automorphism_group(G);
+
+julia> length(gens(qAut));
+168
+```
+"""
+function quantum_automorphism_group(G::Graph)
+  @req typeof(G) == Graph{Undirected} "Graph must be undirected"
+
+  n = nv(G)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n,n)),[2,1])
+
+  nonedges = Tuple{Int,Int}[] # This should be n^2 - |E| many nonedges
+  for i in 1:n 
+    for j in i:n
+      if !has_edge(G,i,j)
+        push!(nonedges,(i,j))
+      end
+    end
+  end
+
+
+  edgs = map(edg -> (src(edg), dst(edg)), collect(edges(G))) # This should be |E| many edges
+
+  new_relations = []
+  function _addrelations(edge,nonedg)
+    r1 = u[edge[1],nonedg[1]] * u[edge[2],nonedg[2]]
+    r2 = u[edge[2],nonedg[1]] * u[edge[1],nonedg[2]]
+    r3 = u[edge[1],nonedg[2]] * u[edge[2],nonedg[1]] 
+    r4 = u[edge[2],nonedg[2]] * u[edge[1],nonedg[1]]
+    append!(new_relations,[r1,r2,r3,r4])
+    return;
+  end
+  for edg in edgs
+    for nonedg in nonedges
+      _addrelations(edg,nonedg)
+    end
+  end
```

Comment:
`new_relations` is a `Vector{Any}` here. You can avoid it e.g. like this:
```suggestion
  function _addrelations(edge,nonedg)
    r1 = u[edge[1],nonedg[1]] * u[edge[2],nonedg[2]]
    r2 = u[edge[2],nonedg[1]] * u[edge[1],nonedg[2]]
    r3 = u[edge[1],nonedg[2]] * u[edge[2],nonedg[1]] 
    r4 = u[edge[2],nonedg[2]] * u[edge[1],nonedg[1]]
    return [r1,r2,r3,r4]
  end
  new_relations = [_addrelations(edg, nonedg) for edg in edgs for nonedg in nonedges]
```

Reply:
Ahh thank you. I solved this differently tho since this suggestion doesn't work.